src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -255,17 +255,44 @@ public E create(E entity, boolean flush) {
     /**
      * @return all entities for a particular type.
      */
-    public CandlepinQuery<E> listAll() {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria();
+    public List<E> listAll() {
+        EntityManager em = this.getEntityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<E> criteriaQuery = criteriaBuilder.createQuery(this.entityType);
+        Root<E> root = criteriaQuery.from(this.entityType);
+        criteriaQuery.select(root);
+
+        List<Predicate> predicates = new ArrayList<>();
+        Predicate securityPredicate = this.getSecurityPredicate(this.entityType, criteriaBuilder, root);
+
+        if (securityPredicate != null) {
+            predicates.add(securityPredicate);
+            criteriaQuery.where(predicates.toArray(new Predicate[predicates.size()]));
+        }
 
```

Comment:
You should be able to just add the security predicate to the `criteria.where` and reduce the following to:

```
        Predicate securityPredicate = this.getSecurityPredicate(this.entityType, criteriaBuilder, root);
        if (securityPredicate != null) {
            criteriaQuery.where(securityPredicate);
        }

```